apps/android/firefox.py

```diff
@@ -54,11 +55,20 @@ def create_certificate(self):
 
     def setup_device(self):
         self.device = ADBAndroid()
-        if self.binary:
-            if self.device.is_app_installed(self.APP_NAME):
-                print("Uninstalling app %s" % self.APP_NAME)
-                self.device.uninstall_app(self.APP_NAME)
-            self.device.install_app(apk_path=self.binary)
+        if self.binary and self.proxy.mode is "record":
+            if not self.app_installed:
```

Comment:
This variable name makes the code difficult to read as we're first checking if the app is installed and then checking it again by querying the device. Perhaps rename it to `skip_install`?

Reply:
renamed